test/TestUtil.js

```diff
@@ -540,3 +540,35 @@ exports.TestFactory = class TestFactory {
         this.clients.clear();
     }
 };
+
+/**
+ * Duplicated this from BuildInfo because the logic of it is changed in 5.1, and in backward compatibility tests older
+ * versions won't be able to access the new logic if we keep the new logic only in src/.
+ */
+exports.calculateServerVersionFromString = (versionString) => {
+    if (versionString == null) {
+        return BuildInfo.UNKNOWN_VERSION_ID;
+    }
+    const mainParts = versionString.split('-');
+    const tokens = mainParts[0].split('.');
+
+    if (tokens.length < 2) {
+        return BuildInfo.UNKNOWN_VERSION_ID;
```

Comment:
Q: Should we copy these constants as well to this function or use them from BuildInfo?

Reply:
I think it should be OK, as these constants are defined on every release but I wouldn't mind copying them either